app/models/profile.rb

```diff
@@ -61,6 +61,26 @@ def canonical?
     parent_profile_id.blank?
   end
 
+  def policy
+    return self unless external
+
+    Profile.includes(:benchmark)
+           .find_by(account: account_id, external: false, ref_id: ref_id,
+                    benchmarks: { ref_id: benchmark.ref_id })
+  end
+
+  def business_objective
+    BusinessObjective.find(business_objective_id) if business_objective_id
+  end
```

Comment:
I'm not sure if there's a better way to do this or how AR does it behind the scenes, but the column name here is actually `business_objective_id`, not `business_objective`.